server/utils/getOutcomeTypesListFilters.js

```diff
@@ -1,14 +1,7 @@
-module.exports = () => {
-  const outcomeTypes = [
-    { label: 'Probation sentence', value: 'PROBATION_SENTENCE' },
-    { label: 'Non-probation sentence', value: 'NON_PROBATION_SENTENCE' },
-    { label: 'Report requested', value: 'REPORT_REQUESTED' },
-    { label: 'Adjourned', value: 'ADJOURNED' },
-    { label: 'Committed to Crown', value: 'COMMITTED_TO_CROWN' },
-    { label: 'Crown plus PSR', value: 'CROWN_PLUS_PSR' },
-    { label: 'No outcome', value: 'NO_OUTCOME' },
-    { label: 'Other', value: 'OTHER' }
-  ]
+const { getOutcomeTypes } = require('../../server/services/case-service')
+
+module.exports = async () => {
+  const outcomeTypes = await getOutcomeTypes()
```

Comment:
Can we not cache this instead of loading static data on every page load? 
```
var outcomeTypes
module.exports = async () => {
   outcomeTypes = outcomeTypes || await getOutcomeTypes()
```
could add some expiry logic. Or set in Redis with expiry. At least save it in session so that it is loaded once per user's session? Just worried if we are adding more overhead to already troubled CCS

Reply:
This is what I said in the first post, the approach is T.D in the making.